CRM/Report/Form/Pledge/Pbnp.php

```diff
@@ -180,9 +194,30 @@ public function select() {
           if (!empty($field['required']) ||
             !empty($this->_params['fields'][$fieldName])
           ) {
-            $select[] = "{$field['dbAlias']} as {$tableName}_{$fieldName}";
-            $this->_columnHeaders["{$tableName}_{$fieldName}"]['type'] = CRM_Utils_Array::value('type', $field);
-            $this->_columnHeaders["{$tableName}_{$fieldName}"]['title'] = CRM_Utils_Array::value('title', $field);
+            if ($fieldName == 'total_paid') {
+              $this->_totalPaid = TRUE; // add pledge_payment join
+              $this->_columnHeaders["{$tableName}_{$fieldName}"] = [
+                'title' => $field['title'],
+                'type' => $field['type'],
+              ];
+              $select[] = "COALESCE(sum({$this->_aliases[$tableName]}.actual_amount), 0) as {$tableName}_{$fieldName}";
+            }
+            elseif ($fieldName == 'balance_due') {
+              $this->_totalPaid = TRUE; // add pledge_payment join
+              $cancelledStatus = array_search('Cancelled', $this->_pledgeStatuses);
```

Comment:
Yep to the above - - this could be 

 $cancelledStatus = CRM_Core_PseudoConstant::getKey('CRM_Pledge_BAO_Pledge', 'status_id', 'Cancelled');